www/include/eventLogs/xml/data.php

```diff
@@ -360,13 +378,23 @@
 
 $whereOutput = "";
 if (isset($output) && $output != "") {
-    $whereOutput = " AND logs.output like '%" . $pearDBO->escape($output) . "%' ";
+    $queryValues['whereOutput'] = [\PDO::PARAM_STR => '%' . CentreonUtils::escapeSecure($output) . '%'];
+    $whereOutput = " AND logs.output like :whereOutput";
 }
 
 $innerJoinEngineLog = "";
 if ($engine == "true" && isset($openid) && $openid != "") {
-    $innerJoinEngineLog = " INNER JOIN instances i ON i.name = logs.instance_name AND i.instance_id IN (" .
-        $pearDBO->escape($openid) . ") ";
+    // filtering poller ids and keeping only real ids
+    $pollerIds = explode(',', $openid);
+    $filteredIds = array_filter($pollerIds, function ($id) {
+        return is_numeric($id);
+    });
+
+    if (count($filteredIds) > 0) {
+        $queryValues['openid'] = [\PDO::PARAM_STR => implode(", ", $filteredIds)];
```

Comment:
This will not work.
You need to bind each poller id separately

Reply:
**Modifications done**